src/fu-engine.c

```diff
@@ -1295,30 +1273,18 @@ fu_engine_install (FuEngine *self,
 	g_autoptr(GHashTable) metadata_hash = NULL;
 
 	g_return_val_if_fail (FU_IS_ENGINE (self), FALSE);
-	g_return_val_if_fail (device_id != NULL, FALSE);
+	g_return_val_if_fail (device != NULL, FALSE);
 	g_return_val_if_fail (AS_IS_STORE (store), FALSE);
 	g_return_val_if_fail (blob_cab != NULL, FALSE);
 	g_return_val_if_fail (error == NULL || *error == NULL, FALSE);
 
-	/* wildcard */
-	if (g_strcmp0 (device_id, FWUPD_DEVICE_ID_ANY) == 0) {
-		device = fu_engine_get_item_by_wildcard (self, store, error);
-		if (device == NULL)
-			return FALSE;
-	} else {
-		/* find the specific device */
-		device = fu_device_list_get_by_id (self->device_list, device_id, error);
-		if (device == NULL)
-			return FALSE;
-	}
-
 	/* check the device is not locked */
 	if (fu_device_has_flag (device, FWUPD_DEVICE_FLAG_LOCKED)) {
 		g_set_error (error,
 			     FWUPD_ERROR,
 			     FWUPD_ERROR_INTERNAL,
 			     "Device %s is locked",
-			     device_id);
+			     fu_device_get_name(device));
```

Comment:
space before bracket * 3 -- also don't want the ID here? It makes it less ambiguous when there is more than one device of the same type.

Reply:
That works for me. I think we have other places that has the ID in square brackets.